src/dev/i18n_tools/tasks/validate_translation_files/remove_outdated_translations.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { I18nCheckTaskContext, MessageDescriptor } from '../../types';
+import { verifyMessageDescriptor } from '../../extractors/formatjs';
+import type { GroupedMessagesByNamespace } from './group_messages_by_namespace';
+import { TaskReporter } from '../../utils/task_reporter';
+export const removeOutdatedTranslations = ({
```

Comment:
Do you think the unused messages could be removed at the same time to avoid processing twice?

Reply:
I prefer not to this keeps them more readable and easier to change in the future and report progress on if needed.
looping over the list twice isnt a performance dealbreaker. The actual checking will be happening anyways so we would be just avoiding looping twice over readablity and maintenance which i wouldnt prefer. The code previously was doing both and it was hard to adapt it to the new changes

Reply:
Thanks for context.